legion/engine/physics/data/colliders/collider_data.hpp

```diff
@@ -17,54 +40,98 @@ namespace legion::physics
+                return { nullptr };
+            }
 
-    class ConvexColliderData : public ColliderData
-    {
-    public:
-        friend class ConvexColliderHandle;
+            return { &m_colliderSpecifics.boxExtents };
+        }
+
+        void setBoxExtents(const math::vec3& newExtents) noexcept;
+        
+        L_ALWAYS_INLINE pointer<const float> getSphereRadius() const 
+        {
+            if (m_colliderType != collider_type::sphere)
+            {
+                log::warn("ColliderData::getSphereRadius called on a collider that is not of type collider_type::sphere");
+                return { nullptr };
+            }
 
-        ConvexColliderData(const math::vec3& offset,const math::quat& rotation, const math::vec3& boxExtents) noexcept;
+            return { &m_colliderSpecifics.sphereRadius };
+        }
 
-        ConvexColliderData(const math::vec3& offset, const math::quat& rotation, InternalConvexColliderPtr ccPtr);
+        void setSphereRadius(float newRadius) noexcept;
+        
+        L_ALWAYS_INLINE InternalConvexColliderPtr getConvexCollider() const 
+        {
+            if (m_colliderType != collider_type::quickhull_convex)
+            {
+                log::warn("ColliderData::getConvexCollider called on a collider that is not of type collider_type::quickhull_convex");
+                return { nullptr };
+            }
 
-        L_ALWAYS_INLINE const math::vec3& getBoxExtents() const noexcept { return m_boxExtents; }
+            return m_colliderSpecifics.internalConvexStructure;
+        }
 
-        L_ALWAYS_INLINE InternalConvexColliderPtr getConvexPtr() const noexcept { return { m_internalConvexStructure }; }
+        L_ALWAYS_INLINE size_type getColliderIndex() const noexcept { return m_colliderIndex; }
 
     private:
 
-        math::vec3 m_boxExtents{0.0f};
-        InternalConvexColliderPtr m_internalConvexStructure{nullptr};
-    };
+        L_ALWAYS_INLINE void setModificationRequestVector(pointer<std::vector<collider_modification_data>> modificationsRequests) noexcept
+        {
+            m_modificationsRequests = modificationsRequests;
+        }
 
-    class SphereColliderData : public ColliderData
-    {
-    public:
+        L_ALWAYS_INLINE void setColliderToBox(const math::vec3& boxExtents)
+        {
+            m_colliderSpecifics.boxExtents = { boxExtents };
+        }
 
-        SphereColliderData(const math::vec3& offset, float radius) noexcept;
+        L_ALWAYS_INLINE void setColliderToConvexCollider(InternalConvexColliderPtr convexMesh)
+        {
+            m_colliderSpecifics.internalConvexStructure = { convexMesh };
+        }
 
-        L_ALWAYS_INLINE const float getRadius() const noexcept { return m_radius; }
+        L_ALWAYS_INLINE void setColliderToSphereCollider(float radius)
+        {
+            m_colliderSpecifics.sphereRadius = { radius };
+        }
```

Comment:
I think templates make sense when the possible types you want to put are unbounded (e.g.  add_component makes sense as a templated function since there are a large number or possible components). But in the case of colliders, there are only four types (box,capsule (later on),sphere, and convex. These functions are also private and not accessible to the user. I would make the argument that turning these into templates would only add unnecessary complexity.